sktime/transformations/series/compose/_conditional_impute.py

```diff
@@ -63,40 +68,8 @@ def transform(self, Z, X=None):
 
         z = check_series(Z, enforce_univariate=True)
 
-        outliers = self.detector_.transform(z)
-
-        Zt = Z.copy()
-        Zt.iloc[outliers[outliers].index] = np.nan
-
-        z = Zt
-        zt = self.imputer_.transform(z)
-        return zt
-
-    def fit_transform(self, Z, X=None):
-        """
-        Fit to training data.
-        Returns a transformed version of Z.
-        Parameters
-        ----------
-        Z : pd.Series
-        X : pd.DataFrame, optional (default=None)
-            Exogenous variables are ignored
-        Returns
-        -------
-        Zt : pd.Series
-            Transformed(Conditionally Imputed) time series.
-        """
-
-        z = check_series(Z, enforce_univariate=True)
-
-        self.detector_ = self.detector.fit(z)
-
-        outliers = self.detector_.transform(z)
-
-        Zt = Z.copy()
-        Zt[outliers[outliers].index] = np.nan
+        outliers = self.annotator_.transform(z)
 
-        z = Zt
-        self.imputer_ = self.imputer.fit(z)
-        zt = self.imputer_.fit_transform(z)
-        return zt
+        z.iloc[outliers[outliers].index] = np.nan
+        z = self.imputer_.transform(z)
```

Comment:
```suggestion
        z = self.imputer_.transform(z, X)
```
We probably also have to subset the X as we do with z above.